src/ds3_uint64_string_map.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2018 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+#include "ds3_uint64_string_map.h"
+
+static void _internal_uint64_free(gpointer data) {
+    g_free((uint64_t*) data);
+}
+
+static void _internal_str_free(gpointer data) {
+    ds3_str_free((ds3_str*) data);
+}
+
+ds3_uint64_string_map* ds3_uint64_string_map_init(void) {
+    // Create a hash table with uint64_t* as the key, and ds3_str* as the value
+    struct _ds3_uint64_string_map* map = g_new0(struct _ds3_uint64_string_map, 1);
+    map->hash = g_hash_table_new_full(g_int64_hash, g_int64_equal, _internal_uint64_free, _internal_str_free);
+    return (ds3_uint64_string_map*) map;
+}
+
+// Inserts a safe copy of the key-value pair into the map. Returns true if the key did not exist yet.
+gboolean ds3_uint64_string_map_insert(ds3_uint64_string_map* map, const uint64_t* key, const ds3_str* value) {
```

Comment:
If this method is called publically, it should not return a gboolean, but a ds3_bool instead.